src/useAsyncOps.js

```diff
@@ -1,37 +1,64 @@
-import { useCallback, useState } from 'react'
+import { useCallback, useState, useContext, useRef } from 'react'
 import { call as callOperation } from './operations'
+import { START, COMPLETE, FAILURE, Context, determineKey } from './Provider'
 
 export default ({
   name,
-  onStart = () => {},
-  onComplete = () => {},
-  onFailure = () => {}
+  onStart = () => { },
+  onComplete = () => { },
+  onFailure = () => { }
 }) => {
-  const [status, setStatus] = useState(false)
+  const { current: hookId } = useRef(Date.now())
+  const [runId, setRunId] = useState()
+  const [status, setStatus] = useState({ loading: false })
```

Comment:
I think it will be easier to understand and and use if you split the status state up into two, loading and error instead of merging them into a single object and mutator

Reply:
But that causes the selector to rerender multiple times.  Doing it this way means the rerender will only happen once when state changes.